### PR TITLE
MM-23596: Ability to list private channels for team

### PR DESCRIPTION
**Pull request by ashishbhate**
#### Summary

store, app, api and go driver support for listing private channels

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-23596

### CODE REVIEW

**Comment by ashishbhate** [comment]
Documentation PR is https://github.com/mattermost/mattermost-api-reference/pull/537

**Comment by reflog** [comment]
@ashishbhate FYI - if you want CI to pass your API documentation checks, you need to push both your branch and the API branch upstream, and not to the fork.

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -1046,6 +1046,35 @@ func (s SqlChannelStore) GetMoreChannels(teamId string, userId string, offset in
 	return channels, nil
 }
 
+func (s SqlChannelStore) GetPrivateChannelsForTeam(teamId string, offset int, limit int) (*model.ChannelList, *model.AppError) {
+	channels := &model.ChannelList{}
+
+	_, err := s.GetReplica().Select(channels, `
```
Comment (reflog) [inline store/sqlstore/channel_store.go:1052]: small nit 1/5 - maybe rewrite this using squirrel? We are trying to keep all non trivial queries written with it

**Comment by ashishbhate** [comment]
@reflog thanks for the info. Another reason for me to stop using my fork of the server.

File: store/sqlstore/channel_store.go (hunk 1 above)
Comment (ashishbhate) [reply store/sqlstore/channel_store.go]: done, switched to squirrel.

**Pull request merged**